### ISSUE TITLE
Generic build crashes on OS X

### ISSUE THREAD

**Issue by rjmooney** [klamonte/qodem#39]
**tl;dr:** I'm able to build under OS X but the binary crashes:

``` sh
$ bin/qodem 
qodem(18810,0x7fff7b18e000) malloc: *** error for object 0x7f976c000000: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

Here's some information on my setup:

``` sh
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.11.5
BuildVersion:   15F34

$ cc --version
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Initially, `configure` was not able to find SDL2 or ncurses. Using the latest homebrew update, I installed them:

``` sh
$ brew install sdl2
$ brew install ncurses
```

Configure was then not able to find `Makefile.in` so to resolve that (and for good measure), I ran:

``` sh
$ aclocal
$ autoconf
$ autoheader
$ automake --add-missing
```

I was then able to configure and build qodem:

``` sh
$ LDFLAGS="-L/usr/local/opt/ncurses/lib -L/usr/local/lib" CPPFLAGS="-I/usr/local/opt/ncurses/include" ./configure
$ make
**lots of warnings here; no errors**
```

However when I run the binary it crashes:

``` sh
$ bin/qodem 
qodem(18810,0x7fff7b18e000) malloc: *** error for object 0x7f976c000000: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

Here are the configure and makefile output:

[config-log.txt](https://github.com/klamonte/qodem/files/355100/config-log.txt)
[make-log.txt](https://github.com/klamonte/qodem/files/355101/make-log.txt)

What am I missing?

**Comment by klamonte** [comment]
Oh gosh, my apologies for missing this bug report!  I had just started preparing to move out my old house and must have missed the notification in the bustle.

Looking at the make-log, a few things to try:

1. --disable-upnp to configure.  miniupnpc is pulling the wrong headers it looks like.
2. 'make -f build/Makefile.generic' and see how the minimal ncurses build compares.

I am seeing several other errors in the make log that might lead to crashes, around the size arguments to snprintf().  I will have to do a more thorough port on a recent OSX / clang.  Plus all warnings from colors.c.  I'll need to noodle on this one.  gcc accepts -finput-charset, but clang does not, I might need to either suppress the warning for clang or put another block of screens in UTF8 encoding for the non-Windows compilers.

Apologies again for the late response. :(

**Comment by klamonte** [comment]
I think we are fixed now, closing.
